core/condition_test.go

```diff
@@ -287,6 +287,27 @@ func TestCondition_evaluate(t *testing.T) {
 			ExpectedSuccess: true,
 			ExpectedOutput:  "[BODY].data.id > 0",
 		},
+		{
+			Name:            "body-jsonpath-hexadecimal-int-using-greater-than",
+			Condition:       Condition("[BODY].data > 0"),
+			Result:          &Result{Body: []byte("{\"data\": \"0x1\"}")},
+			ExpectedSuccess: true,
+			ExpectedOutput:  "[BODY].data > 0",
+		},
```

Comment:
Could you add one more test that makes sure `len([BODY].data) == 3` passes? 

I'm fine with the change, but I'd like to ensure that the length of a string, even if that string is a hexadecimal representation of a number, is treated like a string.

Reply:
Sure. Can I have multiple conditions in these test scenarios, like in the YAML? Or do you prefer new test case? I can add more tests as well, any ideas of what could go wrong here? Maybe longer numbers?

Reply:
Up to you; I'd personally make a separate test case just for `len([BODY].data) == 3`.
No need for testing longer numbers imo.

Reply:
Done! Let me know what you think about it.

Reply:
It's perfect

Reply:
Found an issue. I added a test to compare `[BODY].data == 2` with `{\"data\": \"0x02\"}` and I got this error:

```
--- FAIL: TestCondition_evaluate (0.00s)
    --- FAIL: TestCondition_evaluate/body-jsonpath-hexadecimal-int-using-greater-than-without-strings (0.00s)
        condition_test.go:683: Condition '[BODY].dataa == 2' should have been success=true
        condition_test.go:686: Condition '[BODY].dataa == 2' should have resolved to '[BODY].dataa == 2', got '[BODY].dataa (INVALID) == 2'
```

For some reason, `sanitazeAndResolve` (`core/endpoint.go:232`) fails to extract the hex string from the JSON. I suspect something on the function `walk` in `jsonpath/jsonpath.go` is not right.

Reply:
Oh, actually it was an error on my test case... ignore the previous errors. THe issue is on the isEqual function. It tries to test if `"0x02"` is equal to `"2"`, of course it isn't:

```
--- FAIL: TestCondition_evaluate (0.00s)
    --- FAIL: TestCondition_evaluate/body-jsonpath-hexadecimal-int-using-greater-than-without-strings (0.00s)
        condition_test.go:683: Condition '[BODY].dataa == 2' should have been success=true
        condition_test.go:686: Condition '[BODY].dataa == 2' should have resolved to '[BODY].dataa == 2', got '[BODY].dataa (0x02) == 2'
```

Reply:
Oh, found the issue. Please check my 2 most recent commits. Result of the test suite did not change for me.